src/crt/sllmulu.src

```diff
@@ -0,0 +1,280 @@
+; Performs 48 bit ("short long long") multiplication
+; 
+; Arguments:
+; uiy = Most significant bytes of arg 1
+; ubc = Least significant bytes of arg 1
+; ude = Most signficant bytes of arg 2
+; uhl = Least significant bytes of arg 2
+; 
+; Returns:
+; ude:uhl = ude:uhl * uiy:ubc
+
+	assume	adl=1
+
+	section	.text
+	public	__sllmulu
+__sllmulu:
+	;backup af, alt bc, alt de, and alt hl
```

Comment:
The issue with using shadow registers isn't that they need to be backed up, but that they can be clobbered by interrupts. For a long routine like this, it may be worth disabling interrupts and restoring them at the end (32-bit and 64-bit division in the toolchain already do this).

Reply:
Fixed by Updated sllmulu

Reply:
Sorry, maybe I should have clarified, but when I say "restoring" interrupts, that means checking whether they're enabled or not before disabling them, and then conditionally enabling them at the end. We definitely don't want a multiplication to randomly enable interrupts, after all.